DH2 beta release

kennethbruskiewicz opened the pull request:
Includes all of the changes of #333 and has incorporated the features of #318.

Addresses the following issues:
- #430 
- #431 
- #434 
- #435

Incorporates translated CANCOGEN schema.

The beta release is meant to shorten the feedback loops for the Data Harmonizer. The internationalization features were a "package deal" and so were difficult to roll out as one milestone.

However, a beta release will give users access to partially completed milestones, which will give the developers feedback more often. It will also help them identify bugs more quickly.

This could be also a nightly build which can be shared with users.

ddooley commented:
For these beta releases, to give people a chance to test the stand-alone version, we should run "yarn build:web" command to have the /web/dist/ folder generated and the index.html file there able to load and run DH.  Otherwise they have to be familiar with git environment to get it going, and I think they'd have to run it directly via git too (via yarn dev or yarn build:web)

kennethbruskiewicz committed:
Merge branch 'dh2-with-linkml-schema' into dh2-beta-release

kennethbruskiewicz committed:
re-add 1-M. TODO: loading is too slow with this hook!

kennethbruskiewicz committed:
remove pascalcase workaround. introduce a dict for keeping track of enum text-title correspondences in field

kennethbruskiewicz committed:
use autocomplete renderer

kennethbruskiewicz committed:
WIP: title/text distinction in menus. BUG: multiselect

kennethbruskiewicz committed:
re-add multiselect cell change

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
WIP: title/text distinction in multiselect menus.

ddooley committed:
Updated GRDI specification to v12.2.2 including new 1-many format

ddooley committed:
schema editor update

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
Bugfix: multiple vocabularyTextToTitle merging. WIP: Support for multilingual selection.

kennethbruskiewicz committed:
Workaround: cancogen template merge text and title.

kennethbruskiewicz committed:
WIP: title/text distinction for all schemas

ddooley committed:
GRDI touchup

ddooley committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

ddooley committed:
Cancogen schema update

kennethbruskiewicz committed:
WIP: ROADMAP

kennethbruskiewicz committed:
MILESTONE: title-text distinction with enums and enum exports

kennethbruskiewicz committed:
MILESTONE: title-text distinction with 1-M. WIP: file opening saving for Containers, test data. There is also a bug regarding loading data when data has already been loaded, clearing the values.

kennethbruskiewicz committed:
remove uneccessary script

kennethbruskiewicz committed:
remove WIP dev docs

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

ddooley committed:
french update

ddooley committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

ddooley committed:
documentation diagram

ddooley committed:
Adjusting menus to be handled by name, not title

As well, "del local_choice['text']" wasn't working, and other anciliary non-translation enum parts were being included in french translation so rejigged creating of a locale choice.

ddooley committed:
Fixed {Title_Case} contradicting HostGenderMenu choices

As well, more efficient french locale file

ddooley committed:
Fixing detection of null or undefined object

ddooley committed:
Adjustment to titleOverText() function

Adjustment to titleOverText() function to accept key as a value to return for enumeration choices.  Sometimes "text" attribute is not included as that just reiterates key.

Also warning about missing "location" attribute in json file being loaded.  THIS error detection of malformed json file should be improved.

ddooley committed:
Protecting against missing see_also case

A schema or class might not have a "see_also" attribute
Also there might not be any data in a Container

ddooley committed:
minimal export file.

ddooley committed:
GRDI update to menu naming

ddooley committed:
A bit of programmer-level error trapping.

Though this may also occur when user clicks on some non DH json file.

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
resolve template through schema ID instead of location in container json

kennethbruskiewicz committed:
resolve template through schema ID instead of location in container json

kennethbruskiewicz committed:
resolve template through schema ID instead of location in container json

kennethbruskiewicz committed:
rename excel sheets

kennethbruskiewicz committed:
WIP: transfer between plural and singular forms of classes for containers when saving and loading

kennethbruskiewicz committed:
WIP: transfer between plural and singular forms of classes for containers when saving and loading

kennethbruskiewicz committed:
WIP: transfer between plural and singular forms of classes for containers when saving and loading

ddooley committed:
This allows export.js to be found for this example schema

Its a temporary workaround.  Location shouldn't be relied upon.

ddooley committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

ddooley committed:
Whitespaceminimizedstring caps fix

kennethbruskiewicz committed:
Merge branch 'master' of https://github.com/cidgoh/DataHarmonizer

kennethbruskiewicz committed:
separate out 1m helper functions from context

kennethbruskiewicz committed:
separate out 1m helper functions from context

ddooley committed:
mpox cosmetic update

ddooley committed:
name field used as index in enums

kennethbruskiewicz committed:
listener based crud

kennethbruskiewicz committed:
1-m refactored to module

kennethbruskiewicz committed:
1-m refactored to module with setup function

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-context-cleanup

kennethbruskiewicz committed:
missing context_menu_callback binding

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
didn't pass in proper value  to autocomplete renderer

kennethbruskiewicz committed:
format prettier, small bug in createRowManager

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
remove console log statements

kennethbruskiewicz committed:
remove redundant hot.clear() statements. increases performance and prevents 'afterChange' to be triggered too frequently on large dataset loads.

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
WIP: row-based selection for add-row currentSelection

kennethbruskiewicz committed:
WIP: export sections (no option otherwise. TODO: test import data with sections. test export to TSV/CSV with sections.

kennethbruskiewicz committed:
merged permissible values warning

kennethbruskiewicz committed:
warning message

kennethbruskiewicz committed:
re-add sections to excel exports. empty files export both sections and slot headers.

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-row-selection

kennethbruskiewicz committed:
row selection working over all shared keys

kennethbruskiewicz committed:
row selection with blank

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
new location for getColumnIndexByFieldName

kennethbruskiewicz committed:
follow up selections

kennethbruskiewicz committed:
rename col value to prop

kennethbruskiewicz committed:
refactor consolidate

kennethbruskiewicz committed:
row removal in 1m rework

kennethbruskiewicz committed:
row removal in 1m rework

kennethbruskiewicz committed:
row removal in 1m rework. prettier

ddooley committed:
New schema_editor spec

kennethbruskiewicz committed:
accidental file duplication

kennethbruskiewicz committed:
tweaks

kennethbruskiewicz committed:
console log removal, prettier

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
CSS import order matters?

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
typo

kennethbruskiewicz committed:
refactoring

kennethbruskiewicz committed:
linting and bad var names

kennethbruskiewicz committed:
regression during refactor

kennethbruskiewicz committed:
choosing selection approach

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
Update main.yaml with later node

ddooley committed:
Create dh-validate.py

This is the new validation script for tsv/csv/xlsx/xls files.

ddooley committed:
bug fixed

ddooley committed:
Adding ontology prefixes

ddooley committed:
tweak for better min/max numeric handling

ddooley committed:
prefix update

ddooley committed:
prefix update

ddooley committed:
revert class name change

ddooley committed:
Update dh-validate.py

kennethbruskiewicz committed:
CRUD Support for ACTION.READ, ACTION.UPDATE -> to deletion, to remapping. WIP: selection for current state finicky - needs to update currentSelection in place or draw from calculation. cross-filtering broken

kennethbruskiewicz committed:
READ after UPDATE to ensure selection sync

kennethbruskiewicz committed:
WIP Filter

kennethbruskiewicz committed:
iteration of new filter for testing

kennethbruskiewicz committed:
basic crud with new filter implementation to prevent infinite recursion

kennethbruskiewicz committed:
make columns read only

kennethbruskiewicz committed:
moving code around to comments

kennethbruskiewicz committed:
remove large commented out code on unique key cardinality mapping

kennethbruskiewicz committed:
formatting

kennethbruskiewicz committed:
on focus change needs to be a callback

kennethbruskiewicz committed:
menu ordering

kennethbruskiewicz committed:
formatting

kennethbruskiewicz committed:
WIP: on-row-change validation

kennethbruskiewicz committed:
cleanup console log

kennethbruskiewicz committed:
bug in addRowsToBottom, use null as blank value instead of empty string

ddooley committed:
tweak

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
some logging

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
change to .js export to eliminate jest warnings

kennethbruskiewicz committed:
hotfix: check if unique key has shared slot name before searching for child classes

ddooley committed:
Tweaked GRDI template

kennethbruskiewicz committed:
BUGFIX: block same-table update propagation

kennethbruskiewicz committed:
MILESTONE: only delete on shared key value being final of its kind. TODO: test

kennethbruskiewicz committed:
WIP: reworking template.js tests

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
Remove currying

kennethbruskiewicz committed:
Remove currying

ddooley committed:
tweak GRDI template so it doesn't have outdated dh_interface

ddooley committed:
Update translations.json

ddooley committed:
Enable slot usage to be added to rather than overwritten

ddooley committed:
GRDI Schema update

kennethbruskiewicz committed:
comments

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' of https://github.com/cidgoh/DataHarmonizer into dh2-beta-release

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
parent-child emit/listen crud

kennethbruskiewicz committed:
check currentSelection null

kennethbruskiewicz committed:
WIP: route actions in childListensCRUD

kennethbruskiewicz committed:
MILESTONE: ported 1-level CRUD to parent/child emitter architecture. TODO: read/filter (2-level CRUD)

kennethbruskiewicz committed:
WIP: filtering

kennethbruskiewicz committed:
Filtering with facets in new model

kennethbruskiewicz committed:
restore deletion propagation behaviour

kennethbruskiewicz committed:
gitignore

kennethbruskiewicz committed:
rename ported functions

kennethbruskiewicz committed:
remove deprecated 1m library

kennethbruskiewicz committed:
remove console log statements

kennethbruskiewicz committed:
Merge branch 'master' into dh2-i18n-rc1

kennethbruskiewicz committed:
remove uneccessary imports

kennethbruskiewicz committed:
refactor out buildSchemaTree

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
disable 1-m behaviour

kennethbruskiewicz committed:
remove large bootstrap import

kennethbruskiewicz committed:
reorg imports and remove unecessary assignment

kennethbruskiewicz committed:
reorg imports and remove unecessary assignment

kennethbruskiewicz committed:
hide tabs

kennethbruskiewicz committed:
revert comment

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
Resolve bug with tabs not switching

kennethbruskiewicz committed:
only export non-empty keys when saving in json

kennethbruskiewicz committed:
BUGFIX: empty events with empty keys when selecting entire table

kennethbruskiewicz committed:
BUGFIX: ensure jump to modal and fill to modal refresh contents on active dataharmonizer changes

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-sparse-saving

kennethbruskiewicz committed:
put logs earlier

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
MILESTONE: EventTracker object to manage listeners between template setups to be deleted and refreshed.

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-sparse-saving

kennethbruskiewicz committed:
WIP: translation bugs

kennethbruskiewicz committed:
BUGFIX: language menu not going to the language of the json file

kennethbruskiewicz committed:
rewrite and slim down locale resources function to focus on enums only

kennethbruskiewicz committed:
refactor translation on load code to support new cross-language namespaces. necessary for data loaded from files to be translatable.

kennethbruskiewicz committed:
remove conditional that was preventing loading json data when the source language and schema matched the current interface configuration

kennethbruskiewicz committed:
merge sparse JSON saves with improved language file loading

kennethbruskiewicz committed:
WIP

kennethbruskiewicz committed:
refactor jumpToModal and fillColumnModal for compatibility with translation switching

kennethbruskiewicz committed:
pleasing the linter

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
re-add removeNumericKeys

ddooley committed:
Updated script from master

ddooley committed:
Cancogen menu update

kennethbruskiewicz committed:
Merge branch 'dh2-i18n-rc1' of https://github.com/cidgoh/DataHarmonizer into dh2-i18n-rc1

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
workaround for jumpto/fillto menu sync

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
workaround for jumpto/fillto menu sync

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
BUGFIX: toggle row visibility

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
Merge branch 'master' of https://github.com/cidgoh/DataHarmonizer

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
remove unecessary buttons

kennethbruskiewicz committed:
use latest dh-validate script

kennethbruskiewicz committed:
merge lib/ changes from master

kennethbruskiewicz committed:
console log

kennethbruskiewicz committed:
Merge branch 'dh2-i18n-rc1' of https://github.com/cidgoh/DataHarmonizer into dh2-i18n-rc1

kennethbruskiewicz committed:
WIP test exports

kennethbruskiewicz committed:
remove duplicate and unused fields

kennethbruskiewicz committed:
Merge branch 'master' of https://github.com/cidgoh/DataHarmonizer

kennethbruskiewicz committed:
Merge branch 'master' into dh2-i18n-rc1

kennethbruskiewicz committed:
revert packages.json and web/templates/grdi (for deliberate merge)

kennethbruskiewicz committed:
remove unecessary test

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
add flight check command for pre-build use

kennethbruskiewicz committed:
merge post-master revisions to dh2

kennethbruskiewicz committed:
change flightcheck command to not halt on lint exit

ddooley committed:
GRDI 1-many

ddooley committed:
schema update from main branch

kennethbruskiewicz committed:
Merge branch 'dh2-i18n-rc1' of https://github.com/cidgoh/DataHarmonizer into dh2-i18n-rc1

kennethbruskiewicz committed:
BUGFIX: multiple reloads when template the same accidentally cleared data harmonizers.

kennethbruskiewicz committed:
SCHEMA: SDRF shared ID to Cancogen Schema, causing errors in localization

kennethbruskiewicz committed:
remove console info logs, optimize a loop

kennethbruskiewicz committed:
Merge branch 'dh2-i18n-rc1' into dh2-beta-release

kennethbruskiewicz committed:
remove unused functions

kennethbruskiewicz committed:
remove unused functions, linter

ddooley committed:
grdi and mpox template content fixes

ddooley committed:
Merge branch 'master' into dh2-i18n-rc1

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-i18n-rc1

kennethbruskiewicz committed:
Merge and linter

kennethbruskiewicz committed:
yarn.lock

kennethbruskiewicz committed:
Merge branch 'dh2-i18n-rc1' of https://github.com/cidgoh/DataHarmonizer into dh2-i18n-rc1

kennethbruskiewicz committed:
remove console log

kennethbruskiewicz committed:
remove console log

kennethbruskiewicz committed:
remove redundant calls to translationSelect change

kennethbruskiewicz committed:
linter

kennethbruskiewicz committed:
pushed load json data into conditional which does nothing if the container format isn't adhered to

kennethbruskiewicz committed:
restore tabs to dh2-beta-release

kennethbruskiewicz committed:
remove unecessesary packaging

kennethbruskiewicz committed:
guard against no schema to find templatePath for

kennethbruskiewicz committed:
guard against no schema to find templatePath for

kennethbruskiewicz committed:
use some instead of filter

kennethbruskiewicz committed:
display tabs

kennethbruskiewicz committed:
WIP: popups

kennethbruskiewicz committed:
Merge branch 'master' of https://github.com/cidgoh/DataHarmonizer

kennethbruskiewicz committed:
Merge branch 'master' into dh2-beta-release. Moved back to regular import scheme.

kennethbruskiewicz committed:
Merge branch 'dh2-beta-release' into dh2-1m-rework

kennethbruskiewicz committed:
Linter

kennethbruskiewicz committed:
prettier

kennethbruskiewicz committed:
popup for blocking innapropriate addrows

kennethbruskiewicz committed:
Merge branch 'dh2-1m-rework' into dh2-beta-release

kennethbruskiewicz committed:
merge 1m-rework. addition: popup for blank selections on add rows
